File: app/src/main/java/com/example/data/db/InterviewDao.java
package com.example.data.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
import kotlinx.coroutines.flow.Flow;

@Dao
public interface InterviewDao {

    // User Operations
    @Query("SELECT * FROM users WHERE id = :userId LIMIT 1")
    Flow<UserEntity> getUser(String userId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUser(UserEntity user);

    // Resume Profile Operations
    @Query("SELECT * FROM resume_profiles ORDER BY createdAt DESC")
    Flow<List<ResumeProfileEntity>> getAllResumes();

    @Query("SELECT * FROM resume_profiles WHERE id = :id LIMIT 1")
    ResumeProfileEntity getResumeById(long id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertResume(ResumeProfileEntity resume);

    @Query("DELETE FROM resume_profiles WHERE id = :id")
    void deleteResumeById(long id);

    // Interview Session Operations
    @Query("SELECT * FROM interview_sessions ORDER BY createdAt DESC")
    Flow<List<InterviewSessionEntity>> getAllSessions();

    @Query("SELECT * FROM interview_sessions WHERE id = :sessionId LIMIT 1")
    Flow<InterviewSessionEntity> getSessionById(long sessionId);

    @Query("SELECT * FROM interview_sessions WHERE id = :sessionId LIMIT 1")
    InterviewSessionEntity getSessionSync(long sessionId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertSession(InterviewSessionEntity session);

    @Update
    void updateSession(InterviewSessionEntity session);

    @Query("DELETE FROM interview_sessions WHERE id = :sessionId")
    void deleteSession(long sessionId);

    // Interview Questions Operations
    @Query("SELECT * FROM interview_questions WHERE sessionId = :sessionId ORDER BY questionIndex ASC")
    Flow<List<InterviewQuestionEntity>> getQuestionsForSession(long sessionId);

    @Query("SELECT * FROM interview_questions WHERE sessionId = :sessionId ORDER BY questionIndex ASC")
    List<InterviewQuestionEntity> getQuestionsForSessionSync(long sessionId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertQuestion(InterviewQuestionEntity question);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertQuestions(List<InterviewQuestionEntity> questions);

    @Update
    void updateQuestion(InterviewQuestionEntity question);
}

File: app/src/main/java/com/example/data/db/InterviewDao.kt
package com.example.data.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface InterviewDao {

    // User Operations
    @Query("SELECT * FROM users WHERE id = :userId LIMIT 1")
    fun getUser(userId: String = "default_user"): Flow<UserEntity?>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertUser(user: UserEntity)

    // Resume Profile Operations
    @Query("SELECT * FROM resume_profiles ORDER BY createdAt DESC")
    fun getAllResumes(): Flow<List<ResumeProfileEntity>>

    @Query("SELECT * FROM resume_profiles WHERE id = :id LIMIT 1")
    suspend fun getResumeById(id: Long): ResumeProfileEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertResume(resume: ResumeProfileEntity): Long

    @Query("DELETE FROM resume_profiles WHERE id = :id")
    suspend fun deleteResumeById(id: Long)

    // Interview Session Operations
    @Query("SELECT * FROM interview_sessions ORDER BY createdAt DESC")
    fun getAllSessions(): Flow<List<InterviewSessionEntity>>

    @Query("SELECT * FROM interview_sessions WHERE id = :sessionId LIMIT 1")
    fun getSessionById(sessionId: Long): Flow<InterviewSessionEntity?>

    @Query("SELECT * FROM interview_sessions WHERE id = :sessionId LIMIT 1")
    suspend fun getSessionSync(sessionId: Long): InterviewSessionEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSession(session: InterviewSessionEntity): Long

    @Update
    suspend fun updateSession(session: InterviewSessionEntity)

    @Query("DELETE FROM interview_sessions WHERE id = :sessionId")
    suspend fun deleteSession(sessionId: Long)

    // Interview Questions Operations
    @Query("SELECT * FROM interview_questions WHERE sessionId = :sessionId ORDER BY questionIndex ASC")
    fun getQuestionsForSession(sessionId: Long): Flow<List<InterviewQuestionEntity>>

    @Query("SELECT * FROM interview_questions WHERE sessionId = :sessionId ORDER BY questionIndex ASC")
    suspend fun getQuestionsForSessionSync(sessionId: Long): List<InterviewQuestionEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertQuestion(question: InterviewQuestionEntity): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertQuestions(questions: List<InterviewQuestionEntity>)

    @Update
    suspend fun updateQuestion(question: InterviewQuestionEntity)
}
